allensdk/brain_observatory/behavior/behavior_project_cache/external/behavior_project_metadata_writer.py

```diff
@@ -0,0 +1,320 @@
+        behavior_only_sessions = table.merge(self._release_behavior_only_nwb,
+                                             left_index=True,
+                                             right_index=True)
+
+        # 2) Filter to release ophys sessions (but they get no nwb files)
+        ophys_session_ids = self._get_ophys_sessions_from_ophys_experiments(
+            ophys_experiment_ids=self._release_ophys_experiment_nwb.index
+        )
+        ophys_sessions = table[table['ophys_session_id']
+            .isin(ophys_session_ids)]
+        return pd.concat([behavior_only_sessions, ophys_sessions], sort=False)
+
+    def _get_ophys_session_release_table(
+            self, table: BehaviorOphysSessionsTable) -> pd.DataFrame:
+        """Returns ophys sessions release table
+
+        Parameters
+        ----------
+        table
+            BehaviorOphysSessionsTable
+
+        Returns
+        --------
+        Dataframe including release ophys sessions
+        """
+        table = table.table
+
+        # ophys sessions are different because the nwb files for ophys
+        # sessions are at the experiment level.
+        # We don't want to associate these sessions with nwb files
+        ophys_session_ids = \
+            self._get_ophys_sessions_from_ophys_experiments(
+                ophys_experiment_ids=self._release_ophys_experiment_nwb.index)
+        return table[table.index.isin(ophys_session_ids)]
+
+    def _get_ophys_experiment_release_table(
+            self, table: ExperimentsTable) -> pd.DataFrame:
+        """Returns ophys experiment release table
+
+        Parameters
+        ----------
+        table
+            ExperimentsTable
+
+        Returns
+        --------
+        Dataframe including release ophys experiments with nwb file metadata
+        """
+        return table.table.merge(self._release_ophys_experiment_nwb,
+                                 left_index=True,
+                                 right_index=True)
+
+    def _get_release_files(self, file_type='BehaviorNwb') -> pd.DataFrame:
+        """Gets the release nwb files.
+
+        Parameters
+        ----------
+        file_type
+            NWB files to return ('BehaviorNwb', 'BehaviorOphysNwb')
+
+        Returns
+        ---------
+        Dataframe of release files and file metadata
+            -index of behavior_session_id or ophys_experiment_id
+            -columns file_id and isilon filepath
+        """
+        if file_type not in ('BehaviorNwb', 'BehaviorOphysNwb'):
+            raise ValueError(f'cannot retrieve file type {file_type}')
+
+        if file_type == 'BehaviorNwb':
+            attachable_id_alias = 'behavior_session_id'
+        else:
+            attachable_id_alias = 'ophys_experiment_id'
+
+        query = f'''
+            SELECT attachable_id as {attachable_id_alias}, id as file_id, 
+            filename, storage_directory
+            FROM well_known_files 
+            WHERE published_at IS NOT NULL AND 
```

Comment:
It does not matter for this release, but future releases will probably want to specify a specific date.  The value for this release is '2021-03-25'.